### DISCUSSION TITLE
Vehicles change speed not persisting

### DISCUSSION THREAD

**Discussion by CiscoFrisco** [eclipse/mosaic#113] [Q&A] [answered]
Hello once again,

I'm currently trying to work on an application that changes a vehicle's speed.  In this example, I just stop the vehicle before it enters the intersection and then I try to reset the speed. To stop the vehicle I am calling the method stopNow and it's working fine. To reset the speed back to its original speed I'm using the method changeSpeedWithInterval. I experimented with other methods (resetSpeed, changeSpeedWithPleasantAcceleration) but nothing happened. 

However, as you can see in the video below, right after resetting back the speed, the vehicle just stops. I made some logs to check if it was my application that was forcing the vehicle to stop, but since I only call stopNow once and only one log was shown (for the first time the vehicle stops) the bug isn't probably coming from the application. I wonder, since the vehicle stops exactly at the white line before the intersection, if this is a normal behavior coming from SUMO. 

I would really appreciate any help in this matter. I've been looking through my code for hours trying to find the bug but with no success. Thought it would be better to ask here since this can be something other people may have experienced as well.

https://user-images.githubusercontent.com/32069828/119928574-b8f2db80-bf73-11eb-9344-dbdadcb298c3.mp4

Best regards,
Francisco Filipe

**Comment by kschrab** [comment, marked as answer]
Hi Francisco,

Have you already tried the `resume` command?

The methods you are calling are just translated into SUMO API calls. That is, the `stopNow` command will estimates a position on its current route where the vehicle can be stopped safely without exceed emergency braking. This position is then passed to SUMO using the "stop" command, making the vehicle halt on this very position. The only way to let the vehicle drive again, is to call the `resume` command. The other calls (`changeSpeedWithInterval`, `slowDown`, `resetSpeed`) only work if the vehicle is in "driving" mode.

**Reply by CiscoFrisco** [reply]
Completely missed the resume command. Just tried it and it works. Thanks so much!
